integrations/waffle-mars.md

```diff
@@ -0,0 +1,402 @@
+```typescript
+import { use, expect } from 'chai';
+import { Provider } from '@ethersproject/providers';
+import { solidity } from 'ethereum-waffle';
+import { ethers, Wallet } from 'ethers';
+import { MyToken, MyTokenFactory } from '../build/types';
+
+// Tell Chai to use Waffle's Solidity plugin
+use(solidity);
+
+describe ('MyToken', () => {
+  // Use custom provider to connect to Moonbase Alpha
+  let provider: Provider = new ethers.providers.JsonRpcProvider('https://rpc.testnet.moonbeam.network');
+  let wallet: Wallet;
+  let walletTo: Wallet;
+  let token: MyToken;
+
+  beforeEach(async () => {
+    // Logic for setting up the wallet and deploying MyToken will go here
+  });
+
+  // Tests will go here
+})
+```
+
+3. Before each test is run, you'll want to create wallets and connect them to the provider, use the wallets to deploy an instance of the `MyToken` contract, and then call the `initialize` function once with an initial supply of 10 tokens:
+```typescript
+  beforeEach(async () => {
+    const PRIVATE_KEY = '<insert-your-private-key-here>'
+    // Create a wallet instance using your private key & connect it to the provider
+    wallet = new Wallet(PRIVATE_KEY).connect(provider);
+
+    // Create a random account to transfer tokens to & connect it to the provider
+    walletTo = Wallet.createRandom().connect(provider);
+
+    // Use your wallet to deploy the MyToken contract
+    token = await new MyTokenFactory(wallet).deploy();
+
+    // Mint 10 tokens to the contract owner, which is you
+    let contractTransaction = await token.initialize(10);
+
+    // Wait until the transaction is confirmed before running tests
+    await contractTransaction.wait();
+  });
+```
+
+4. Now you can create your first test. The first test will check your initial balance to ensure you received the initial supply of 10 tokens. However, to follow good testing practices, write a failing test first:
+```typescript
+it('Mints the correct initial balance', async () => {
+  expect(await token.balanceOf(wallet.address)).to.equal(1); // This should fail
+});
+```
+
+5. Before you can run your first test, you'll need to go back to the root direction and add a `.mocharc.json` Mocha configuration file:
+```
+cd .. && touch .mocharc.json
+```
+
+6. Now edit the `.mocharc.json` file to configure Mocha:
+```json
+{
+  "require": "ts-node/register/transpile-only", // Use ts-node to transpile the code for tests
+  "timeout": 600000, // Set timeout to 10 minutes
+  "extension": "test.ts" // Specify extension for test files
+}
+```
+
+7. You'll also need to add a script in the `package.json` to run your tests:
+```json
+"scripts": {
+  "build": "waffle",
+  "test": "mocha",
+},
+```
+
+8. You're all set to run the tests, simply use the `test` script you just created and run:
+```
+npm run test
+```
+Please note that it could take a few minutes to process, but if all worked as expected, you should have one failing test.
```

Comment:
I would state here that is because tests are being ran against Moonbase Alpha. But you can run them against a Moonbeam development node.